iogt/middleware.py

```diff
@@ -0,0 +1,27 @@
+from django.conf import settings
+from django.http import HttpResponsePermanentRedirect
+
+
+class SSLRedirectMiddleware(object):
+    def process_request(self, request):
+        HTTPS_PATHS = getattr(settings, 'HTTPS_PATHS', [])
+        response_should_be_secure = self.response_should_be_secure(
+            request, HTTPS_PATHS)
+        request_is_secure = self.request_is_secure(request)
+        if response_should_be_secure and not request_is_secure:
+            return HttpResponsePermanentRedirect(
+                "https://{}{}".format(request.get_host(),
+                                      request.get_full_path())
+            )
+
+    def response_should_be_secure(self, request, HTTPS_PATHS):
+        for path in HTTPS_PATHS:
+            if request.path.startswith(u'/{}'.format(path)):
```

Comment:
Is `request.path` a string (I don't know if there's some Django magic here)? If it is, `startswith` is not going to necessarily match parent paths only, e.g. both `/admin` and `/admin-django` could be matched by `/admin` which may be unexpected. The solution would probably be to parse the path into some kind of proper URL abstraction.

Also, personally I'd just require the `HTTPS_PATHS` setting values to start with `/`--or this would be another thing a URL library would take care of. If somebody _did_ put a path in the settings with a leading `/` it wouldn't work as expected currently.

_Also_, tiny nit... having an uppercase parameter name is weird to me 😅.